loader/src/main/java/net/neoforged/fml/loading/moddiscovery/AbstractModProvider.java

```diff
@@ -34,30 +35,28 @@ public abstract class AbstractModProvider implements IModProvider
     protected static final String MANIFEST = "META-INF/MANIFEST.MF";
 
     protected IModLocator.ModFileOrException createMod(Path... path) {
-        var mjm = new ModJarMetadata();
-        var sj = SecureJar.from(
-                Manifest::new,
-                jar -> jar.moduleDataProvider().findFile(MODS_TOML).isPresent() ? mjm : JarMetadata.from(jar, path),
-                null,
-                path
-        );
+        var jarContents = new JarContentsBuilder()
+                .paths(path)
+                .ignoreRootPackages("assets", "data")
```

Comment:
Note: this skips scanning the `assets` and `data` folders of mods for classes... :)